nexus/blueprint-execution/src/omicron_zones.rs

```diff
@@ -0,0 +1,346 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Manges deployment of Omicron zones to Sled Agents
+
+use anyhow::Context;
+use futures::stream;
+use futures::StreamExt;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::lookup::LookupPath;
+use nexus_db_queries::db::DataStore;
+use nexus_types::deployment::OmicronZonesConfig;
+use sled_agent_client::Client as SledAgentClient;
+use slog::info;
+use slog::warn;
+use slog::Logger;
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+/// Idempotently ensure that the specified Omicron zones are deployed to the
+/// corresponding sleds
+pub async fn deploy_zones(
```

Comment:
I'm tempted to say this should be `pub(crate)` since the only caller is the main `realize_blueprint` entry point, but I don't feel strongly about it (e.g., if you think there might be a reason to call this directly from some other crate's test suite or something).

Reply:
👍 done in cdc88d0b7.